Source/Core/Heap.js

```diff
@@ -0,0 +1,200 @@
+/*global define*/
+define([
+        './Check',
+        './defaultValue',
+        './defined',
+        './defineProperties'
+    ], function(
+        Check,
+        defaultValue,
+        defined,
+        defineProperties) {
+    'use strict';
+
+    /**
+     * Array implementation of a heap.
+     *
+     * @alias Heap
+     * @constructor
+     * @private
+     *
+     * @param {Function} comparator The comparator to use for the heap. If comparator(a, b) is less than 0, sort a to a lower index than b, otherwise sort to a higher index.
+     */
+    function Heap(comparator) {
+        //>>includeStart('debug', pragmas.debug);
+        Check.defined('comparator', comparator);
+        //>>includeEnd('debug');
+
+        this._comparator = comparator;
```

Comment:
In general, you should always expose any constructor parameters as properties (in this case a public getter would be fine).  People should always be able to get back out of an object what they put into it.